lib/refile/s3.rb

```diff
@@ -60,12 +67,14 @@ def initialize(region:, bucket:, max_size: nil, prefix: nil, hasher: Refile::Ran
     verify_uploadable def upload(uploadable)
       id = @hasher.hash(uploadable)
 
-      if uploadable.is_a?(Refile::File) and uploadable.backend.is_a?(S3) and uploadable.backend.access_key_id == access_key_id
-        object(id).copy_from(copy_source: [@bucket_name, uploadable.backend.object(uploadable.id).key].join("/"))
+      operation, options = if upload_via_copy_operation?(uploadable)
+        [:copy_from, { copy_source: [@bucket_name, uploadable.backend.object(uploadable.id).key].join("/") }]
       else
-        object(id).put(body: uploadable, content_length: uploadable.size)
+        [:put, { body: uploadable, content_length: uploadable.size }]
       end
 
+      object(id).send(operation, s3_options_for(operation, options))
```

Comment:
I think this is a bit too clever, and I'm not really sure why it's needed. Can't we just add the options to how it was before?